Ex5/GraphNode.java

```diff
@@ -0,0 +1,30 @@
+public class GraphNode <T> {
```

Comment:
This feels super dangerous. You are inserting these objects into a hashmap, and a hashset, but your object does not implement .equals() or .hashcode(), meaning equality isn't defined for your domain - it becomes object equality.

If you implement equals and hashcode() you will be able to write more of the code without safety checks.